File: ios/RunningApp/Models/Post.swift
import Foundation

// MARK: - 动态帖子
struct Post: Codable, Identifiable {
    let id: Int
    let userId: Int
    let user: User?
    let content: String
    let images: [String]?
    let recordId: Int?
    let record: RunningRecord?
    let likeCount: Int
    let commentCount: Int
    let isLiked: Bool
    let createdAt: Int64
    let updatedAt: Int64

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case user, content, images
        case recordId = "record_id"
        case record
        case likeCount = "like_count"
        case commentCount = "comment_count"
        case isLiked = "is_liked"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

// MARK: - 评论
struct Comment: Codable, Identifiable {
    let id: Int
    let postId: Int
    let userId: Int
    let user: User?
    let content: String
    let replyToUserId: Int?
    let replyToUser: User?
    let createdAt: Int64

    enum CodingKeys: String, CodingKey {
        case id
        case postId = "post_id"
        case userId = "user_id"
        case user, content
        case replyToUserId = "reply_to_user_id"
        case replyToUser = "reply_to_user"
        case createdAt = "created_at"
    }
}

File: android/app/src/main/java/com/runningapp/data/remote/model/Post.kt
package com.runningapp.data.remote.model

import com.google.gson.annotations.SerializedName

/**
 * 动态帖子
 */
data class Post(
    @SerializedName("id")
    val id: Int,
    @SerializedName("user_id")
    val userId: Int,
    @SerializedName("user")
    val user: User?,
    @SerializedName("content")
    val content: String,
    @SerializedName("images")
    val images: List<String>?,
    @SerializedName("record_id")
    val recordId: Int?,
    @SerializedName("record")
    val record: RunningRecord?,
    @SerializedName("like_count")
    val likeCount: Int,
    @SerializedName("comment_count")
    val commentCount: Int,
    @SerializedName("is_liked")
    val isLiked: Boolean,
    @SerializedName("created_at")
    val createdAt: Long,
    @SerializedName("updated_at")
    val updatedAt: Long
)

/**
 * 评论
 */
data class Comment(
    @SerializedName("id")
    val id: Int,
    @SerializedName("post_id")
    val postId: Int,
    @SerializedName("user_id")
    val userId: Int,
    @SerializedName("user")
    val user: User?,
    @SerializedName("content")
    val content: String,
    @SerializedName("reply_to_user_id")
    val replyToUserId: Int?,
    @SerializedName("reply_to_user")
    val replyToUser: User?,
    @SerializedName("created_at")
    val createdAt: Long
)
